### ISSUE TITLE
compilation error with JsonDeserialize 

### ISSUE THREAD

**Issue by DavidLeoni** [immutables/immutables#117]
Hi,

playing with @JsonDeserialize and Immutables both 2.0.11 and 2.1-SNAPSHOT 
 for this class:

```

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.util.List;
import org.immutables.value.Value;

@Value.Immutable(singleton=true,
                builder = false)
@JsonDeserialize(as=ImmutableMyClass.class)
abstract class MyClass {                
    public abstract List<String> prop();    
}
```
I get compilation error
```
ImmutableMyClass.java:[108,35] error: non-static method prop() cannot be referenced from a static context
```

The culprit generated method is this:
```
  @JsonCreator
  public static ImmutableMyClass fromAllAttributes(
      @JsonProperty("prop") List<String> prop) {
    ImmutableMyClass instance = ImmutableMyClass.of();
    if (prop != null) {
      instance = instance.withProp(prop());
    }
    return instance;
  }
```
because the line
```
      instance = instance.withProp(prop());
```
contains `prop()` , but it should be just `prop`

Also, if I declare `prop()` to return just `String` instead of `List<String>` I get at the line of declaration:

```
it/davidleoni/bugreports/TestBug.java:[11,20] error: cannot find symbol
```
which is rather confusing.

**Comment by elucash** [comment]
Thank you for the report! It's clearly reproducible and will be fixed in a nearest release. As as workaround in a meanwhile you can enable `(builder = true)` just for the sake of deserialization to work properly, (it will still canonicalize to the singleton in case `prop` have no values). After the fix you will be able to switch builder off again.

**Comment by elucash** [comment]
Thank you, this should be fixed in version 2.0.12 available on the central
